From acb4670711ac35091567575a79bbafb8a08d4570 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Thu, 25 Aug 2022 18:05:43 +0200
Subject: [PATCH] dt-bindings: interrupt-controller: stm32-exti: add property
 'wakeup-parent'

Some EXTI interrupts can be routed to a parent controller able to
wakeup the SoC from suspend.

Add the optional property 'wakeup-parent' to EXTI binding to
specify the wakeup capable interrupt controller.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I8628a57dbc3dc939ba8a6fd52ab090a47c08cac1
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/383195
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 .../bindings/interrupt-controller/st,stm32-exti.yaml          | 4 ++++
 1 file changed, 4 insertions(+)

--- a/Documentation/devicetree/bindings/interrupt-controller/st,stm32-exti.yaml
+++ b/Documentation/devicetree/bindings/interrupt-controller/st,stm32-exti.yaml
@@ -42,6 +42,8 @@ properties:
     description:
       Interrupts references to primary interrupt controller
 
+  wakeup-parent: true
+
 required:
   - "#interrupt-cells"
   - compatible
@@ -97,9 +99,11 @@ examples:
         interrupt-controller;
         #interrupt-cells = <2>;
         reg = <0x5000d000 0x400>;
+        wakeup-parent = <&pwr_irq>;
         interrupts-extended =
             <&intc GIC_SPI 6 IRQ_TYPE_LEVEL_HIGH>,
             <0>,
+            <&pwr_irq 0 IRQ_TYPE_EDGE_FALLING 0>,
             <&intc GIC_SPI 9 IRQ_TYPE_LEVEL_HIGH>;
     };
 
